modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/AssignmentsTracker.java

```diff
@@ -82,22 +76,33 @@ public AssignmentsTracker(VaultManager vaultManager, MetaStorageManager msManage
     public void startTrack() {
         msManager.registerPrefixWatch(ByteArray.fromString(STABLE_ASSIGNMENTS_PREFIX), assignmentsListener);
 
-        try (Cursor<VaultEntry> cursor = vaultManager.range(
-                ByteArray.fromString(STABLE_ASSIGNMENTS_PREFIX),
-                ByteArray.fromString(incrementLastChar(STABLE_ASSIGNMENTS_PREFIX))
-        )) {
-            for (VaultEntry entry : cursor) {
-                String key = entry.key().toString();
+        msManager.recoveryFinishedFuture().thenAccept(recoveryRevision -> {
+            try (Cursor<Entry> cursor = msManager.getLocally(ByteArray.fromString(STABLE_ASSIGNMENTS_PREFIX),
```

Comment:
This code looks unsightly. If the recovery future should to be waited before local range request is executed it have to be pammening inside of getLocally method. Otherwise, why do we wait for the future?

Reply:
I wait it here in order to retrieve the recoveryRevision. This future is guaranteed to be completed here. It's a common practice, e.g. check TableManager#start